### DISCUSSION TITLE
What counts as Kustomize application with parameter overrides?

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [argoproj/argo-cd#17644] [Q&A] [answered]
I'm looking at concurrent processing section here https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/#enable-concurrent-processing. It says that for Kustomize applications with parameter overrides there's no workaround for now to enable parallel processing. I wonder what counts as such. Is it about the Kubernetes files for the app itself, or also applies to Kubernetes files of app parts, like ConfigMaps, Deployments etc.? If the parameter overrides were already rendered by a deploy scripts and put in a dedicated repository ArgoCD pulls from, would it still be counted? How does ArgoCD determine that some value has a parameter override? I saw examples of special comments used to instruct deploy scripts to replace placeholders with values. Should those comments be deleted in the rendered version of the manifest?
Thank you.

**Comment by crenshaw-dev** [comment, marked as answer]
"Parameter overrides" means parameters in this block: https://github.com/argoproj/argo-cd/blob/442dac12a7d49cc4e120882f4a8283089120b4df/docs/operator-manual/application.yaml#L94-L112

The Argo CD repo-server sets those parameters before running `kustomize build` by editing kustomization.yaml in the manifest base directory.

If more than one process edits that file at the same time, it could cause unpredictable `kustomize build` behavior. 

So if anything in the `spec.source.kustomize` block modifies kustomization.yaml, we take out a lock to prevent other processes from using the edited kustomization.yaml.

**Reply by andrii-korotkov-verkada** [reply]
Thank you!
